sources/api/netdog/src/networkd/config/network.rs

```diff
@@ -121,3 +138,477 @@ impl Display for DhcpBool {
+        match dhcp4 {
+            Dhcp4ConfigV1::DhcpEnabled(b) => *b,
+            Dhcp4ConfigV1::WithOptions(o) => o.enabled,
+        }
+    }
+
+    fn dhcp4_required(dhcp4: &Dhcp4ConfigV1) -> bool {
+        match dhcp4 {
+            // Assume enabled == required
+            Dhcp4ConfigV1::DhcpEnabled(enabled) => *enabled,
+            // If "optional" isn't set, assume DHCP is required.
+            // If optional==true, DHCP is NOT required
+            Dhcp4ConfigV1::WithOptions(o) => o.optional.map_or(true, |b| !b),
+        }
+    }
+
+    fn dhcp4_metric(dhcp4: &Dhcp4ConfigV1) -> Option<u32> {
+        match dhcp4 {
+            Dhcp4ConfigV1::DhcpEnabled(_) => None,
+            Dhcp4ConfigV1::WithOptions(o) => o.route_metric,
+        }
+    }
+
+    fn dhcp6_enabled(dhcp6: &Dhcp6ConfigV1) -> bool {
+        match dhcp6 {
+            Dhcp6ConfigV1::DhcpEnabled(b) => *b,
+            Dhcp6ConfigV1::WithOptions(o) => o.enabled,
+        }
+    }
+
+    fn dhcp6_required(dhcp6: &Dhcp6ConfigV1) -> bool {
+        match dhcp6 {
+            // Assume enabled == required
+            Dhcp6ConfigV1::DhcpEnabled(enabled) => *enabled,
+            // If "optional" isn't set, assume DHCP is required
+            // If optional==true, DHCP is NOT required
+            Dhcp6ConfigV1::WithOptions(o) => o.optional.map_or(true, |b| !b),
+        }
+    }
+}
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::networkd::config::tests::{builder_data, test_data, TestDevices};
+    use crate::networkd::devices::{NetworkDBond, NetworkDInterface, NetworkDVlan};
+
+    fn network_path(name: String) -> PathBuf {
+        test_data()
+            .join("network")
+            .join(format!("{}.network", name))
+    }
+
+    fn network_from_interface(iface: NetworkDInterface) -> NetworkConfig {
+        let mut network = NetworkBuilder::new_interface(iface.name);
+        network.with_dhcp(iface.dhcp4, iface.dhcp6);
+        iface.static4.map(|s| network.with_static_config(s));
+        iface.static6.map(|s| network.with_static_config(s));
+        iface.routes.map(|r| network.with_routes(r));
+        network.build()
+    }
+
+    fn network_from_vlan(vlan: NetworkDVlan) -> NetworkConfig {
+        let mut network = NetworkBuilder::new_vlan(vlan.name);
+        network.with_dhcp(vlan.dhcp4, vlan.dhcp6);
+        vlan.static4.map(|s| network.with_static_config(s));
+        vlan.static6.map(|s| network.with_static_config(s));
+        vlan.routes.map(|r| network.with_routes(r));
+        network.build()
+    }
+
+    fn network_from_bond(bond: NetworkDBond) -> NetworkConfig {
+        let mut network = NetworkBuilder::new_bond(bond.name.clone());
+        network.with_dhcp(bond.dhcp4, bond.dhcp6);
+        bond.static4.map(|s| network.with_static_config(s));
+        bond.static6.map(|s| network.with_static_config(s));
+        bond.routes.map(|r| network.with_routes(r));
+        network.build()
+    }
```

Comment:
This logic will most likely go away once it's integrated into the device code.  It exists to enable unit testing now before that code is available.